package/MDAnalysis/coordinates/chain.py

```diff
@@ -561,17 +557,12 @@ def _read_frame(self, frame):
         # update Timestep
         self.ts = self.active_reader.ts
         self.ts.frame = frame  # continuous frames, 0-based
+        self._current_frame = frame
         return self.ts
 
     def _chained_iterator(self):
         """Iterator that presents itself as a chained trajectory."""
         self._rewind()  # must rewind all readers
-        for i in range(self.n_frames):
-            j, f = self._get_local_frame(i)
-            self.__activate_reader(j)
-            self.ts = self.active_reader[f]
-            self.ts.frame = i
-            yield self.ts
```

Comment:
Then delete all `_chained_reader()` – at least in the current form it does not make much sense.